research/object_detection/core/preprocessor.py

```diff
@@ -739,6 +738,10 @@ def random_pixel_value_scale(image,
     image: image which is the same shape as input image.
   """
   with tf.name_scope('RandomPixelValueScale', values=[image]):
+    image = tf.convert_to_tensor(image, name='image')
+    # Remember original dtype to so we can convert back if needed
+    orig_dtype = image.dtype
```

Comment:
From line 728 [above](https://github.com/tensorflow/models/pull/2146/files#diff-8edc6c706ec1c943f095159ba7116e6cR728), image is already a float32 tensor so these convertions are not necessary.

You can either modify the function description or add a seperate function to deal with multiple input dtype.

Reply:
I removed the conversion as it is not useful.

At the time of the writing I didn't see the conversion [here](https://github.com/tensorflow/models/blob/master/research/object_detection/trainer.py#L65). I think the main problem is that the image is in float32 but with range [0,255] so that i doesn't play well with clipping to [0,1] or tf.image.adjust_* ops.

I updated the commit with scaling by / 255 because tf.image.adjust_* need that (see the doc of [tf.image.rgb_to_hsv](https://www.tensorflow.org/api_docs/python/tf/image/rgb_to_hsv) used by tf.image.adjust_hue)

I can add a clipping to [0, 255] if you want.